lang/java/reef-tang/tang/src/main/java/org/apache/reef/tang/implementation/java/AvroClassHierarchy.java

```diff
@@ -0,0 +1,200 @@
+package org.apache.reef.tang.implementation.java;
+
+import org.apache.reef.tang.ClassHierarchy;
+import org.apache.reef.tang.annotations.NamedParameter;
+import org.apache.reef.tang.exceptions.NameResolutionException;
+import org.apache.reef.tang.formats.avro.*;
+import org.apache.reef.tang.implementation.types.*;
+import org.apache.reef.tang.types.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Implementation of ClassHierarchy for Avro
+ */
+public class AvroClassHierarchy implements ClassHierarchy {
```

Comment:
Thanks for the suggestion @jwang98052 and @markusweimer! Moving the `AvroClassHierarchy` and `AvroClassHierarchySerializer` into the `implementation/avro` package makes sense. However, that would be a short-term solution: As @markusweimer mentioned, having multiple `ClassHierarchy` implementations does not seem to be a good design. I think we can apply the short-term solution at the moment, and file another jira for integrating `ClassHierarchy` implementation. How does it sound?